java/src/test/java/linalg/algorithms/GramSchmidtTest.java

```diff
@@ -59,17 +61,28 @@ void DecomposeIdentityMatrix() {
       for (int j = 0; j < 3; j++)
         dmb.SetValue(i, j, arr[i][j]);
 
-    GramSchmidt res = new GramSchmidt(dmb.BuildMatrix());
-    res.Decompose();
-    Matrix Q = res.Q();
-    Matrix R = res.R();
-
-    for (int i = 0; i < 3; i++) {
-      for (int j = 0; j < 3; j++) {
-        Assertions.assertEquals(arr[i][j], Q.ValueAt(i, j), 0.01);
-        Assertions.assertEquals(arr[i][j], R.ValueAt(i, j), 0.01);
-      }
-    }
+    gram_schmidt.Decompose(dmb.BuildMatrix());
+    Matrix Q = gram_schmidt.GetQ();
+    Matrix R = gram_schmidt.GetR();
+
+    Matrix id = MatrixFactory.DenseMultiply(Q, Q.Transpose());
+    Matrix res = MatrixFactory.DenseMultiply(Q, R);
+
+    Assertions.assertEquals(id.Rows(), id.Cols());
+    Assertions.assertEquals(arr.length, res.Rows());
+    Assertions.assertEquals(arr[0].length, res.Cols());
+
+    for (int i = 0; i < id.Rows(); i++)
```

Comment:
Please make a separate function that would compare 2 Matrices with a certain tolerance (epsilon). It will ba rather useful. You can see that you're doing the same thing (matrix comparison entry by entry) a lot of times here. This is code repetition, which is a big no-no, use a function to avoid it.